libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -1671,32 +1672,71 @@ impl NodeAPI for Greenlight {
+            .channels;
 
         // Ensure one private channel from each peer.
-        open_channels.dedup_by_key(|c| c.peer_id.clone());
+        open_peer_channels.dedup_by_key(|c| c.peer_id.clone());
+        channels.dedup_by_key(|c| c.source.clone());
 
-        // Ceate a routing hint from each channel.
-        for c in open_channels {
-            let (alias_remote, _) = match c.alias {
+        // Create a routing hint from each channel.
+        for peer_channel in open_peer_channels {
+            let peer_id = peer_channel.clone().peer_id.unwrap_or_default();
+            let (alias_remote, _) = match peer_channel.alias {
                 Some(a) => (a.remote.clone(), a.local.clone()),
                 None => (None, None),
             };
 
-            let optional_channel_id = alias_remote.clone().or(c.short_channel_id.clone());
+            let optional_channel_id = alias_remote
+                .clone()
+                .or(peer_channel.short_channel_id.clone());
 
             if let Some(channel_id) = optional_channel_id {
+                // The local fee policy
+                let peer_fee_base_msat = peer_channel.fee_base_msat.unwrap_or_default().msat as u32;
+                let peer_fees_proportional_millionths =
+                    peer_channel.fee_proportional_millionths.unwrap_or_default();
+                // The remote fee policy
+                let channel = channels.clone().into_iter().find(|c| c.source == peer_id);
+                let fees_base_msat = channel.clone().map(|c| c.base_fee_millisatoshi);
+                let fees_proportional_millionths = channel.map(|c| c.fee_per_millionth);
+                debug!(
+                    "For peer {:?}: local base {:?} proportional {:?}, remote base {:?} proportional {:?}",
+                    peer_id,
+                    peer_fee_base_msat,
+                    peer_fees_proportional_millionths,
+                    fees_base_msat,
+                    fees_proportional_millionths
+                );
                 let scid = parse_short_channel_id(&channel_id)?;
                 let hint = RouteHint {
                     hops: vec![RouteHintHop {
-                        src_node_id: hex::encode(c.peer_id.ok_or(anyhow!("no peer id"))?),
+                        src_node_id: hex::encode(
+                            peer_channel.peer_id.ok_or(anyhow!("no peer id"))?,
+                        ),
                         short_channel_id: scid,
-                        fees_base_msat: c.fee_base_msat.clone().unwrap_or_default().msat as u32,
-                        fees_proportional_millionths: c
-                            .fee_proportional_millionths
-                            .unwrap_or_default(),
+                        fees_base_msat: fees_base_msat.unwrap_or(peer_fee_base_msat),
+                        fees_proportional_millionths: fees_proportional_millionths
+                            .unwrap_or(peer_fees_proportional_millionths),
```

Comment:
If the remote fee policy cannot be found, we should not include the route hint in the response. It should never make sense to include the local channel policy in a route hint.

Reply:
Ok, I changed it so that it only adds route hints where the source channel is found